draft-ietf-dnsop-domain-verification-techniques.md

```diff
@@ -170,13 +187,15 @@ The RECOMMENDED format is application-specific underscore prefix labels. Domain
 
 ### Random Token {#random-token}
 
-A unique token used in the challenge. It should be a random value with the following properties:
+A unique token used in the challenge. It should be a random value issued to the user by the provider with the following properties:
 
 1. MUST have at least 128 bits of entropy.
-2. base64url ({{!RFC4648, Section 5}}) encoded.
+2. base64url ({{!RFC4648, Section 5}}) encoded or base16 ({{!RFC4648, Section 8}}) encoded.
 
 See {{RFC4086}} for additional information on randomness requirements.
 
+Hexadecimal base16 encoding is recommended when the random token would exist in a DNS label such as in a CNAME target.
```

Comment:
should this be normative language i.e. RECOMMENDED?

Reply:
Also I think we should provide some rationale here, even if brief.

Reply:
I guess it depends on whether the CNAME can resolve to a hostname, which inherits the LDH (letters digits hyphens) restriction. In that case, base64 alphabet violates the hostname rules, but base16 does not.

So, it depends on what the application provider uses the CNAME to point to.

Reply:
But, lets's see if Erik has any additional rationale or justification for the use of this alphabet.

Reply:
```suggestion
Hexadecimal base16 encoding is RECOMMENDED when the random token would exist in a DNS label such as in a CNAME target.  This is because base64 relies mixed case (and DNS is case-insensitive as clarified in {{RFC4343}}) and because some base64 characters ("/", "+", and "=") may not be permitted by implementations that limit allowed characters to those allowed in hostnames.
```

How is this?  We'd also need to add RFC4343 to the informational references.

Reply:
Ok, yes, I think I can see a stronger rationale for such restrictions then. Technically, a domain name (as long as it isn't a hostname) permits arbitrary 8-bit characters. But yeah, provisioning systems may be likely to limit the character set, so perhaps a more restricted alphabet for domain name labels here will help avoid those problems.

Reply:
There are mechanisms such as case randomization in queries that assume a lack of case-sensitivity and are also deployed to mitigate Kaminsky attacks (https://lists.dns-oarc.net/pipermail/dns-operations/2023-January/021947.html and https://datatracker.ietf.org/doc/html/draft-vixie-dnsext-dns0x20-00).  Between RFC4343 and these trying to rely on mixed-case in labels seems like asking for trouble.  (There's an inherent conflict between the "permits arbitrary 8-bit characters" and the "case insensitive" of rfc4343.)

It may make sense to merge this overall PR #86 and then refine the details of the recommended encoding in a separate PR?  (I'm starting to envision merge conflicts with overlaps against this.)